src/common/hist_util.cc

```diff
@@ -67,34 +68,85 @@ void HistCutMatrix::Init(DMatrix* p_fmat, uint32_t max_num_bins) {
   // Use group index for weights?
   bool const use_group_ind = num_groups != 0 && weights.size() != info.num_row_;
 
-  for (const auto &batch : p_fmat->GetRowBatches()) {
-    size_t group_ind = 0;
-    if (use_group_ind) {
-      group_ind = this->SearchGroupIndFromBaseRow(group_ptr, batch.base_rowid);
+  if (use_group_ind) {
```

Comment:
I replaced std::vector by SimpleArray because std::vector has to slow resize() function, it fill memory by very not performance manner.
Amortization of  allocation/deallocation is not needed here, because I need to call resize() only one time.